jdk-annotations/test-annotated/build.gradle

```diff
@@ -0,0 +1,67 @@
+plugins {
+    id "java-library"
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(21)
+    }
+}
+
+dependencies {
+    compileOnly deps.apt.autoService
+    annotationProcessor deps.apt.autoService
+    compileOnly project(":nullaway")
+    api deps.build.jspecify
+    implementation project(":jdk-annotations:astubx-generator-cli")
+
+    // Add the jdk-javac-plugin as annotation processor
+    annotationProcessor files("${rootProject.projectDir}/jdk-javac-plugin/build/libs/jdk-javac-plugin-all.jar")
+}
+
+def jsonOutputDir = file("${buildDir}/generated/json")
+def astubxOutputDir = file("${buildDir}/generated/astubx")
+
+tasks.withType(JavaCompile).configureEach { compileTask ->
+    if(name == "compileJava") {
+        outputs.dir(jsonOutputDir)
+        dependsOn(":jdk-javac-plugin:shadowJar")
+
+        doFirst {
+            jsonOutputDir.mkdirs()
+
+            options.compilerArgs += [
+                "-Xplugin:NullnessAnnotationSerializer ${jsonOutputDir.absolutePath}"
+            ]
+        }
+    }
+}
+
+// generate .astubx files from JSON
+task generateAstubx(type: JavaExec) {
+    group = "build"
+    description = "Generate .astubx files from JSON using AstubxGeneratorCLI"
+
+    def javaToolchains = project.extensions.getByType(JavaToolchainService)
+    javaLauncher = javaToolchains.launcherFor(java.toolchain)
+
+    inputs.dir(jsonOutputDir)
+    outputs.dir(astubxOutputDir)
+
+    doFirst {
+        astubxOutputDir.mkdirs()
+    }
+
+    mainClass = "com.uber.nullaway.jdkannotations.AstubxGeneratorCLI"
+    classpath = sourceSets.main.runtimeClasspath +
+            project(":jdk-annotations:astubx-generator-cli").sourceSets.main.runtimeClasspath
```

Comment:
This looks weird.  I feel like we should just be executing the shadowJar generated from the cli project.  You can see how we do this in JarInfer [here](https://github.com/uber/NullAway/blob/master/jar-infer/test-java-lib-jarinfer/build.gradle#L28-L28) which I think is a bit cleaner.